File: 0136_Single_Number/136.py
def singleNumber(nums):
    for i in nums:
        if nums.count(i) == 1:
            return i
print(singleNumber(nums = [2,2,1]))
print(singleNumber(nums = [4,1,2,1,2]))
print(singleNumber(nums = [1]))

File: 0136_Single_Number/136.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int temp = nums[0];
        
        for (int i = 1; i < nums.size(); i += 2) {
            if (nums[i] != temp) {
                return temp;
            }
            temp = nums[i + 1];
        }
        return temp;
    }
};

int main() {
    vector<int> v = {4, 1, 2, 1, 2};
    Solution a;
    cout << a.singleNumber(v) << "\n";
    return 0;
}